src/modem/sim800.c

```diff
@@ -185,13 +184,34 @@ static int sim800_attach(struct cellular *modem)
         "AT+CMEE=2",                    /* Enable extended error reporting. */
         "AT+CLTS=0",                    /* Don't sync RTC with network time, it's broken. */
         "AT+CIURC=0",                   /* Disable "Call Ready" URC. */
+        "AT+ICF=3,3",                   /* Disable Parity control (odd)*/
         "AT&W0",                        /* Save configuration. */
         NULL
     };
     for (const char *const *command=init_strings; *command; command++)
         at_command_simple(modem->at, "%s", *command);
 
-    /* Configure IP application. */
+    /* Physical channel has to be reconfigure the same as gsm module*/
+    at_set_parity(modem->at,PARITY_ODD);
+    switch(at_get_parity(modem->at))
```

Comment:
I would use tcgetattr() to determine what parity is set by Nuttx and then on this basis decide how to configure modem.